elcid/assets/js/elcid/services/exception_handler.js

```diff
@@ -0,0 +1,13 @@
+angular.module('opal.services').factory('$exceptionHandler', function($injector, $log) {
+    "use strict";
+    var errorCatcherHandler = function errorCatcherHandler(exception, cause) {
+        // we use the injector to stop a circular dependency as per
+        // https://stackoverflow.com/questions/22332130/injecting-http-into-angular-factoryexceptionhandler-results-in-a-circular-de
+        var $http = $injector.get("$http");
+        var url = "/elcid/v0.1/error_emailer/"
+        $http.post(url, {exception: exception, cause: cause});
```

Comment:
How sure are we that an uncaught HTTP failure / 404/500 won't drop into an infinite loop ?

Reply:
so a 500 doesn't cause an infinite loop. Even if the error emailer itself throws a 500 we don't go infinite. So I think we're good

Reply:
What if there's an error in the call to $http.post somewhere though ?

Reply:
It all I _think_ works fine, for example if you don't pass in a url to a get request it raises an error but only for that request, we don't tumble into an infinite loop. It all works as one would hope. ie pinging the error emailer etc.